code/Control/UserDefinedFormController.php

```diff
@@ -268,6 +274,12 @@ public function process($data, $form)
                             return;
                         }
 
+                        // generate image thumbnail to show in asset-admin
+                        // you can run userforms without asset-admin, so need to ensure asset-admin is installed
+                        if (class_exists(AssetAdmin::class) && $file->getIsImage()) {
+                            Injector::inst()->get(AssetAdmin::class)->generateThumbnails($file);
```

Comment:
```suggestion
                            AssetAdmin::singleton()->generateThumbnails($file);
```

This does the same thing and you get type-hinting in your IDE.

Reply:
Updated